extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerStorageParameters.java

```diff
@@ -58,11 +58,13 @@
    */
   private static final long MINIMUM_SUPER_SORTER_TMP_STORAGE_BYTES = 1_000_000_000L;
 
+  private final boolean isIntermediateStorageLimitConfigured;
   private final long intermediateSuperSorterStorageMaxLocalBytes;
 
-  private WorkerStorageParameters(final long intermediateSuperSorterStorageMaxLocalBytes)
+  private WorkerStorageParameters(final long intermediateSuperSorterStorageMaxLocalBytes, final boolean isIntermediateStorageLimitConfigured)
```

Comment:
Can we do this without the new class parameter `isIntermediateStorageLimitConfigured`. 

if someone sets `intermediateSuperSorterStorageMaxLocalBytes` = -1 we treat the value as not set. 
Meaning instead of the class parameter you would have a method which says 

```
 public boolean isIntermediateStorageLimitConfigured()
  {
    return !intermediateSuperSorterStorageMaxLocalBytes==-1;
  }
  ```

Reply:
I initially did it this way since the ByteTracker still needs a default value(Long.MAX) if it is disabled. However, now I think it would be neater to move that check there instead to avoid the parameter.

Reply:
Updated